renderlib/AppScene.h

```diff
@@ -45,41 +46,45 @@ class Lighting
 public:
   Lighting(void)
     : m_NoLights(0)
+    , m_Lights{ nullptr, nullptr, nullptr, nullptr }
+    , m_sceneLights{ nullptr, nullptr, nullptr, nullptr }
   {
   }
-
-  Lighting& operator=(const Lighting& Other)
+  ~Lighting()
   {
-    for (int i = 0; i < MAX_NO_LIGHTS; i++) {
-      m_Lights[i] = Other.m_Lights[i];
-      m_sceneLights[i].m_light = &m_Lights[i];
+    for (int i = 0; i < MAX_NO_LIGHTS; ++i) {
+      delete m_sceneLights[i];
+      delete m_Lights[i];
     }
-
-    m_NoLights = Other.m_NoLights;
-
-    return *this;
   }
 
-  void AddLight(const Light& Light)
+  Light& LightRef(int i) const { return *m_Lights[i]; }
+
+  void AddLight(Light& light)
   {
     if (m_NoLights >= MAX_NO_LIGHTS)
       return;
 
-    m_Lights[m_NoLights] = Light;
-    m_sceneLights[m_NoLights].m_light = &m_Lights[m_NoLights];
+    m_Lights[m_NoLights] = new Light(light);
+    m_sceneLights[m_NoLights] = new SceneLight(m_Lights[m_NoLights]);
 
     m_NoLights = m_NoLights + 1;
   }
-
-  void Reset(void)
+  void SetLight(int i, Light& light)
   {
-    m_NoLights = 0;
-    // memset(m_Lights, 0 , MAX_NO_LIGHTS * sizeof(CLight));
+    if (i >= MAX_NO_LIGHTS)
+      return;
+
+    delete m_sceneLights[i];
+    delete m_Lights[i];
+
+    m_Lights[i] = new Light(light);
+    m_sceneLights[i] = new SceneLight(m_Lights[i]);
   }
 
-  Light m_Lights[MAX_NO_LIGHTS];
+  Light* m_Lights[MAX_NO_LIGHTS];
   int m_NoLights;
-  SceneLight m_sceneLights[MAX_NO_LIGHTS];
+  SceneLight* m_sceneLights[MAX_NO_LIGHTS];
```

Comment:
a key change here (and in the lines above) is that the light sources are now stored and handled as pointers instead of raw objects.   For convenience, in some places we access them as pointers and in other places as references.  This can be neatened up in another PR (maybe using smart pointers) but doesn't clutter this change set too much.